datalad/interface/run.py

```diff
@@ -0,0 +1,268 @@
+            return
+
+        if rerun:
+            # pull run info out of the last commit message
+            err_info = get_status_dict('run', ds=ds)
+            if not ds.repo.get_hexsha():
+                yield dict(
+                    err_info, status='impossible',
+                    message='cannot re-run command, nothing recorded')
+                return
+            last_commit_msg = ds.repo.repo.head.commit.message
+            cmdrun_regex = r'\[DATALAD RUNCMD\] (.*)=== Do not change lines below ===\n(.*)\n\^\^\^ Do not change lines above \^\^\^'
+            runinfo = re.match(cmdrun_regex, last_commit_msg, re.MULTILINE | re.DOTALL)
+            if not runinfo:
+                yield dict(
+                    err_info, status='impossible',
+                    message='cannot re-run command, last saved state does not look like a recorded command run')
+                return
+            rec_msg, runinfo = runinfo.groups()
+            if message is None:
+                # re-use commit message, if nothing new was given
+                message = rec_msg
+            try:
+                runinfo = json.loads(runinfo)
+            except Exception as e:
+                yield dict(
+                    err_info, status='error',
+                    message=('cannot re-run command, command specification is not valid JSON: %s', e.message))
+                return
+            if 'cmd' not in runinfo:
+                yield dict(
+                    err_info, status='error',
+                    message='cannot re-run command, command specification missing in recorded state')
+                return
+            cmd = runinfo['cmd']
+            rec_exitcode = runinfo.get('exit', 0)
+            rel_pwd = runinfo.get('pwd', None)
+            if rel_pwd:
+                # recording is relative to the dataset
+                pwd = normpath(opj(ds.path, rel_pwd))
+            else:
+                rel_pwd = None  # normalize, just in case
+                pwd = None
+
+            # now we have to find out what was modified during the last run, and enable re-modification
+            # ideally, we would bring back the entire state of the tree with #1424, but we limit ourself
+            # to file addition/not-in-place-modification for now
+            to_unlock = []
+            for r in ds.diff(
+                    recursive=True,
+                    revision='HEAD~1...HEAD',
+                    return_type='generator',
+                    result_renderer=None):
+                if r.get('type', None) == 'file' and \
+                        r.get('state', None) in ('added', 'modified'):
+                    r.pop('status', None)
+                    to_unlock.append(r)
+            if to_unlock:
+                for r in ds.unlock(to_unlock, return_type='generator', result_xfm=None):
+                    yield r
+        else:
+            # not a rerun, figure out where we are running
+            pwd = getpwd()
```

Comment:
such distinctions would not fit into my memory... why not just to stay consistent with the rest of the API, which currently says that e.g. all paths, if `-d` provided, are from the top of that dataset.  So it only makes it all consistent to use for `run` to dictate to run the command from the top of the dataset.  Also would make it easier to e.g. sweep through all datasets which have consistent layout (e.g. BIDS) to cause some modification in all of them (e.g. `for ds in ds*; do datalad run -d $ds do_something_to_bids_dataset path_inside; done`)